### ISSUE TITLE
[bug] gitian development setup - osslsigncode 1.7.1 gives 404 error from sourceforge

### ISSUE THREAD

**Issue by fmhc** [dogecoin/dogecoin#2369]
# Bug Report

**Current behavior:**
the setup information stated in doc/release-process.md which is linked from doc/gitian-building.md is not straightforward and has some issues. 
Currently I am facing problems with fetching the osslsigncode part, as this 
```bash
wget -P inputs http://downloads.sourceforge.net/project/osslsigncode/osslsigncode/osslsigncode-1.7.1.tar.gz
```
will fail with an 404 error.
There are multiple forks on github, which one to choose? Please discuss

**Expected behavior**
archive found and downloaded to proceed with gitian setup

**Steps to reproduce:**
followed instructions to build gitian 

**Screenshots**
![image](https://user-images.githubusercontent.com/3387385/124691007-61814d00-dedb-11eb-9945-0de60934b32d.png)

**Dogecoin Core version**
1.14.4-dev

**Machine specs**
- OS: Windows running a VM with Debian
- CPU: i5 'nough speed
- RAM: 16 GB
- Disk size: 1 TB
- Disk Type (HD/SDD): SSD

**Extra information**

**Comment by AbcSxyZ** [comment]
Possibly using http://security.ubuntu.com/ubuntu/pool/universe/o/osslsigncode/, I had the 1.71 version when I tried on a vm, not sure where I found it.

Good luck for gitian, I failed on my first try personally :smile: 
I'm wondering how core devs do it actually, few signers, idk if they have an old environment available, or if there is another way to do it. 
Depending on the situation, I guess more up-to-date dependencies could be used.

**Comment by patricklodder** [comment]
Confirm the 404. It looks like the `.orig` file on the ubuntu repo is the one we would want but I personally use ad-hoc gitian machines that I build from scratch with every release (per [this](https://gist.github.com/patricklodder/fcffff8af0df78bdfb2957581c7368e5), and I just used [this](https://gist.github.com/patricklodder/88d6c4e3406db840963f85d95ceb44fe) for testing #2333). @rnicoll / @langerhans could either of you provide a sha256sum of your tarball?

**Comment by micaelmalta** [comment]
This has been fixed on the 1.14.4-dev branch: https://github.com/dogecoin/dogecoin/blob/1.14.4-dev/contrib/gitian-build.sh

**Comment by patricklodder** [comment]
> This has been fixed on the 1.14.4-dev branch

Haven't seen anyone checking the shasums in #2343, so let's do that here - add the checks to the script maybe (?) - and we're good 😊

**Comment by michilumin** [comment]
So umm, the contents of my /gitian-builder/inputs/osslsigncode-1.7.1.tar.gz ... is SourceForge's website.  As in, the osslsigncode repo seems to be gone from sourceforge (from what I can find) and instead the 'We can't find that project, but here's some things you may enjoy' website gets downloaded as that file.

**Comment by micaelmalta** [comment]
@michilumin @patricklodder @rnicoll 

Hello, I worked on a way to use Gitian very easily and using only Docker: it s based on @patricklodder gist: https://gist.github.com/patricklodder/88d6c4e3406db840963f85d95ceb44fe

https://github.com/micaelmalta/gitian-builder-dogecoin

Checksum for dependencies like osslsigncode have been fixed and also introduce the checksum

Enjoy!

**Comment by patricklodder** [comment]
I saw a compare from @rnicoll on discord that the `osslsigncode_1.7.1.orig.tar.gz` on ubuntu security and launchpad match sha256sum for the original sourceforge tarball. Although I'd like the proof to be documented, I'll just assume this to be safe for now and we can move on w/ using the launchpad one. 

Can we please implement sha256sum checks on these downloads in contrib/gitian-build.sh? I'll build them into my gists too.

**Comment by micaelmalta** [comment]
@patricklodder we can see the old sha256 sum here: https://github.com/dogecoin/dogecoin/blob/master/contrib/gitian-descriptors/gitian-win-signer.yml#L22 that exactly match the one from launchpad

`f9a8cdb38b9c309326764ebc937cba1523a3a751a7ab05df3ecc99d18ae466c9`

**Comment by patricklodder** [comment]
> that exactly match the one from launchpad

Thanks! Yes, that works!

**Comment by patricklodder** [comment]
This was now fully merged into docs and gitian descriptors under 1.14.4-dev - closing this.
